addons/crm/crm_phonecall.py

```diff
@@ -136,109 +99,111 @@ def schedule_another_phonecall(self, cr, uid, ids, schedule_time, call_summary,
                     'description' : call.description or False,
                     'date' : schedule_time,
                     'team_id' : team_id or False,
-                    'partner_id': call.partner_id and call.partner_id.id or False,
+                    'partner_id' : call.partner_id and call.partner_id.id or False,
                     'partner_phone' : call.partner_phone,
                     'partner_mobile' : call.partner_mobile,
                     'priority': call.priority,
                     'opportunity_id': call.opportunity_id and call.opportunity_id.id or False,
             }
-            new_id = self.create(cr, uid, vals, context=context)
+            new_id = self.create(vals)
             if action == 'log':
-                self.write(cr, uid, [new_id], {'state': 'done'}, context=context)
+                new_id.write({'state': 'done'})
             phonecall_dict[call.id] = new_id
         return phonecall_dict
 
-    def _call_create_partner(self, cr, uid, phonecall, context=None):
-        partner = self.pool.get('res.partner')
-        partner_id = partner.create(cr, uid, {
-                    'name': phonecall.name,
-                    'user_id': phonecall.user_id.id,
-                    'comment': phonecall.description,
+    @api.one
+    def _call_create_partner(self):
+        partner_id = self.partner_id.create({
+                    'name': self.name,
+                    'user_id': self.user_id.id,
+                    'comment': self.description,
                     'address': []
         })
-        return partner_id
+        return partner_id.id
 
-    def on_change_opportunity(self, cr, uid, ids, opportunity_id, context=None):
-        values = {}
-        if opportunity_id:
-            opportunity = self.pool.get('crm.lead').browse(cr, uid, opportunity_id, context=context)
-            values = {
-                'team_id' : opportunity.team_id and opportunity.team_id.id or False,
-                'partner_phone' : opportunity.phone,
-                'partner_mobile' : opportunity.mobile,
-                'partner_id' : opportunity.partner_id and opportunity.partner_id.id or False,
-            }
-        return {'value' : values}
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        self.partner_phone = self.partner_id.phone
+        self.partner_mobile = self.partner_id.mobile
+
+    @api.onchange('opportunity_id')
+    def on_change_opportunity(self):
+        if self.opportunity_id:
+            self.team_id = self.opportunity_id.team_id and self.opportunity_id.team_id.id or False
```

Comment:
no need to check self.opportunity_id.team_id  and self.opportunity_id.team_id.id .. for recordset if no value it's return null recordset.can you check?